compiler/modules/nback.llvm/llvm.bal

```diff
@@ -235,9 +235,18 @@ public class FunctionDefn {
         return label;
     }
 
-    function genReg() returns string {
-        string reg = "%" + "R" + self.varCount.toString();
-        self.varCount += 1;
+    function genReg(string? name = ()) returns string {
```

Comment:
To be completely bullet proof here, you need deal with the case where the name passed to genReg ends  with a `.`  followed by a number

Reply:
Fixed